docs(CMC): What is CMC? (#5501)
dfinity/portal: docs/references/system-canisters/index.mdx

## Cycles minting

> Often referred to as the CMC (short for cycles minting canister).

Reviewer: ```suggestion
The cycles minting canister (CMC) has canister ID [`rkp4c-7iaaa-aaaaa-aaaca-cai`](https://dashboard.internetcomputer.org/canister/rkp4c-7iaaa-aaaaa-aaaca-cai).
```

> The CMC converts ICP to [cycles](/docs/building-apps/essentials/gas-cost). Cycles are used up by canisters as they do work (simply storing data counts as doing work, even when no code in the canister is running). In this conversion, ICP is destroyed, and cycles are created. Thus, this operation is referred to as cycles "minting".

Reviewer: ```suggestion
Cycles are used by canisters to pay for the resources they use, such as data storage, compute power, or network bandwidth. The CMC converts ICP into [cycles](/docs/building-apps/essentials/gas-cost) by burning ICP and creating cycles. This operation is referred to as cycles "minting."
```

Author: How about we swap the order of the first two sentences in your suggestion?

Reviewer: IMO it makes more sense to define cycles first before describing how the CMC uses cycles.

> The CMC has canister ID [`rkp4c-7iaaa-aaaaa-aaaca-cai`](https://dashboard.internetcomputer.org/canister/rkp4c-7iaaa-aaaaa-aaaca-cai).

Reviewer: ```suggestion

```

The CMC canister is hosted on the system subnet [`tdb26-jop6k-aogll-7ltgs-eruif-6kk7m-qpktf-gdiqx-mxtrf-vb5e6-eqe`](https://dashboard.internetcomputer.org/subnet/tdb26-jop6k-aogll-7ltgs-eruif-6kk7m-qpktf-gdiqx-mxtrf-vb5e6-eqe).

> In general, you must do a couple things to use this canister:
> 
> 1. Send ICP to the CMC (by calling the ICP ledger canister).
> 2. Tell the CMC about the ICP that was sent.
> 

Reviewer: ```suggestion
### Using the CMC 

To use the CMC, you must first send ICP to the CMC by calling the ICP ledger canister, then tell the CMC about the ICP that was sent.
```

When minting cycles, the ICP transfer parameters must meet the following requirements:

> 1. `subaccount` - Matches the recipient of the new cycles. More on this below.
> 2. `memo` - Indicates what type of operation will be performed using the ICP.

Reviewer: ```suggestion
1. `subaccount`: The recipient of the new cycles. More on this below.
2. `memo`: Indicates what type of operation will be performed using the ICP.
```

The subaccount corresponding to a recipient principal is constructed as follows:

> 1. Start with an arry of length 32 with all elements set to 0.
> 2. The first byte is set to the length of the principal (as a blob, not in the string format).
> 3. The principal bytes are copied into the remaining bytes of the subaccount.

Reviewer: ```suggestion
1. Start with an array of length `32` with all elements set to `0`.
2. The first byte is set to the length of the principal as a `blob`.
3. The principal bytes are copied into the remaining bytes of the subaccount.
```

For example, to send 4.2 ICP to CMC such that `principal "mcj7w-qq3pa-zsxyt-fjs6f-4s7px-mrq3c-i6wmj-ciqw6-7q4aj-odhgc-4ae"` will end up getting the newly created cycles, ICP could be sent to the CMC like this:

```bash
CYCLES_MINTING_CANISTER_ID=rkp4c-7iaaa-aaaaa-aaaca-cai
ICP_LEDGER_CANISTER_ID=ryjl3-tyaaa-aaaaa-aaaba-cai

dfx canister --ic call $ICP_LEDGER_CANSITER_ID icrc1_transfer "(record {
  to = record {
    owner = principal "${CYCLES_MINTING_CANISTER_ID}";

    // This corresponds to principal "mcj7w-qq3pa-zsxyt-fjs6f-4s7px-mrq3c-i6wmj-ciqw6-7q4aj-odhgc-4ae".
    subaccount = opt blob "\1d\1b\78\33\2b\e2\65\4c\bc\5e\4b\ef\bb\23\0d\89\1e\b3\12\24\42\de\fc\38\04\b8\67\30\b8\02\00\00";
  };

  // This matches what the CMC's notify_mint_cycles method requires.
  memo = opt blob "\4d\49\4e\54\00\00\00\00";

  // 4.2 ICP, denominated in 1/100_000_000ths of of an ICP, known as an e8.
  amount = 420_000_000 : nat;

  // Also in e8s. Transfers are very inexpensive in ICP :D
  fee = opt (10_000 : nat);
})"
```

> This transaction will be recorded in a block whose index is printed by the above command. For example, suppose the block index is `123_456_789`. This is used in step 2, which can be performed as follows:

Reviewer: ```suggestion
This transaction will be recorded in a block whose index is returned by the above command. For example, suppose the block index is `123_456_789`. This index used in the next step:
```

```bash
dfx canister --ic call $CYCLES_MINTING_CANISTER_ID notify_mint_cycles '(record {
  block_index = 123_456_789 : nat64;
})'
```

> After the above command, the destination principal has a higher cycles balance. To read the balance, the cycles ledger canister can be queried like so:

Reviewer: ```suggestion
The destination principal now has a higher cycles balance. To read the balance, the cycles ledger canister can be queried:
```

```bash
CYCLES_LEDGER_CANISTER_ID=um5iw-rqaaa-aaaaq-qaaba-cai

dfx canister --ic call $CYCLES_LEDGER icrc1_balance_of '(record {
    owner = principal "mcj7w-qq3pa-zsxyt-fjs6f-4s7px-mrq3c-i6wmj-ciqw6-7q4aj-odhgc-4ae";
})'
```

> The exchange rate from ICP to cycles varies, depending on the price of ICP in XDR (see the exchange rate canister). The result is that the price of cycles in XRD is relatively stable.

Reviewer: ```suggestion
The exchange rate from ICP to cycles varies, depending on the price of ICP in XDR (see the [exchange rate canister](/docs/references/system-canisters/xrc). 
```

Author: I think the last sentence contains an important point. Otherwise, it's not clear what is the point of varying exchange rate between ICP and cycles.
